src/code_transformation/util.jl

```diff
@@ -110,3 +110,52 @@ replace_vararg(typ::SymOrExpr, vararg_info::Tuple) =
         vararg_info[2] == :no_N || vararg_info[2] == :Vararg ?
             replace_body(typ, :(Vararg{$(get_body(typ))})) :
             replace_body(typ, :(Vararg{$(get_body(typ)), $(vararg_info[2])}))
+
+"""
+    parse_kwargs(nt_expr) -> NamedTuple
+
+Accepts an expression containing a `NamedTuple` literal and parses it into a `NamedTuple`
+with expressions as values.
+"""
+function parse_kwargs(nt_expr)
+    if isempty(nt_expr.args)
+        return NamedTuple()
+    end
+
+    first_arg = first(nt_expr.args)
+    if first_arg.head == :parameters
+        return parse_kwargs_parameters(nt_expr)
+    elseif first_arg.head == :(=)
+        return parse_kwargs_tuple(nt_expr)
+    else
+        throw(ArgumentError("Unsupported expression $nt_expr for kwargs;"
+            * " they must be passed as a NamedTuple literal"))
+    end
+end
+
+function parse_kwargs_tuple(tup_expr)
+    nt_names = Tuple(first(ex.args) for ex in tup_expr.args)
+    return NamedTuple{nt_names}(last(ex.args) for ex in tup_expr.args)
+end
+
+function parse_kwargs_parameters(param_tuple_expr)
+    param_expr = param_tuple_expr[1]
+
+    nt_names = Tuple(first(ex.args) for ex in param_expr.args)
+    return NamedTuple{nt_names}(last(ex.args) for ex in param_expr.args)
+end
+
+"""
+    parse_is_node(bool_array_expr) -> Vector{Bool}
+
+Accepts an expression containing a `Vector{Bool}` literal and parses it into a
+`Vector{Bool}`.
+"""
+function parse_is_node(bool_array_expr)
+    if bool_array_expr.head != :vect
+        throw(ArgumentError("Unsupported expression $bool_array_expr for is_node; "
+            * "it must be passed as a `Vector{Bool}` literal (e.g., `[true, false]`)"))
+    end
+
+    return Bool[arg for arg in bool_array_expr.args]
```

Comment:
```suggestion
    return Bool.(bool_array_expr.args)
```

Reply:
Yours creates a `BitVector` and I want `Vector{Bool}`. I could use `map` if you like.

Reply:
I think `map(Bool, bool_array_expr.args)` is cleaner than `convert(Vector{Bool}, bool_array_expr.args)`

Reply:
There's also `collect(Bool, bool_array_expr.args)` which is identical in behaviour to the original in the case where the constructor and `convert` method sets are different.